packages/grid/_modules_/grid/hooks/features/core/gridState.ts

```diff
@@ -28,8 +28,8 @@ export interface GridState {
 
 export const getInitialState: () => GridState = () => ({
   rows: getInitialRowState(),
-  pagination: { ...INITIAL_PAGINATION_STATE },
-  options: { ...DEFAULT_GRID_OPTIONS },
+  pagination: INITIAL_PAGINATION_STATE,
+  options: DEFAULT_GRID_OPTIONS,
```

Comment:
The state will be part of the public API, from what I understand from #190. It's important that the state is **immutable**. If there are leaks, we need to find and fix them. This will help spot them, it also removes runtime work that shouldn't be necessary.